service/secret_service.go

```diff
@@ -36,13 +36,22 @@ func (s secretService) Save(ctx context.Context, projectName string, namespaceNa
 		return NewError(models.SecretEntity, ErrInvalidArgument, "secret name cannot be empty")
 	}
 
-	// TODO: Add new service method to get only project and namespace id for names
-	namespaceSpec, err := s.nsService.Get(ctx, projectName, namespaceName)
-	if err != nil {
-		return err
+	var projectSpec models.ProjectSpec
+	namespaceSpec := models.NamespaceSpec{}
+	var err error
+	if namespaceName == "" { // Namespace is optional for secrets
```

Comment:
we don't need to fetch the projectSpec & namespaceSpec and do the save, why don't we just pass the namespace name & projectname to the repo layer & repo layer handles saving it by ID.

Reply:
It will change the design of entire secret repo, maybe I can do that as part of separate pr.